platforms/avr/timer.c

```diff
@@ -111,27 +105,16 @@ inline uint32_t timer_read32(void) {
  * FIXME: needs doc
  */
 inline uint16_t timer_elapsed(uint16_t last) {
-    uint32_t t;
-
-    ATOMIC_BLOCK(ATOMIC_RESTORESTATE) {
-        t = timer_count;
-    }
-
-    return TIMER_DIFF_16((t & 0xFFFF), last);
+    return TIMER_DIFF_16(timer_read(), last);
 }
 
 /** \brief timer elapsed32
  *
+
  * FIXME: needs doc
  */
 inline uint32_t timer_elapsed32(uint32_t last) {
-    uint32_t t;
-
-    ATOMIC_BLOCK(ATOMIC_RESTORESTATE) {
-        t = timer_count;
-    }
-
-    return TIMER_DIFF_32(t, last);
+    return TIMER_DIFF_32(timer_read32(), last);
 }
```

Comment:
These functions can actually become generic instead of duplicated for all platforms (their implementation now becomes identical across all platforms).  However, making them actually inlined would probably increase the code size (on AVR a 32-bit comparison is a lot of instructions).

Reply:
I had made a version like this before, because all implementations were the same, moving the implementation to timer.h (or timer.c, if we don't want to inline them), but later I had one version that kept the AVR code unchanged, but I decided to move forward with this and forgot to deduplicate them.

It didn't make the AVR size larger, though. It stayed the same, IIRC.